internal/db/schema/migrations/postgres/16/01_plugin.up.sql

```diff
@@ -0,0 +1,149 @@
+  comment on function insert_plugin_subtype() is
+    'insert_plugin_subtype() inserts sub type name into the base type plugin table';
+
+  create or replace function
+    update_plugin_subtype()
+    returns trigger
+  as $$
+  begin
+    update plugin set name = new.name where public_id = new.public_id and new.name != name;
+    return new;
+  end;
+  $$ language plpgsql;
+  comment on function update_plugin_subtype() is
+    'update_plugin_subtype() will update base plugin type name column with new values from sub type';
+
+  -- delete_plugin_subtype() is an after delete trigger function
+  -- for subtypes of plugin
+  create or replace function delete_plugin_subtype()
+    returns trigger
+  as $$
+  begin
+    delete from plugin
+    where public_id = old.public_id;
+    return null; -- result is ignored since this is an after trigger
+  end;
+  $$ language plpgsql;
+  comment on function delete_plugin_subtype is
+    'delete_plugin_subtype() is an after trigger function for subytypes of plugin';
+
+  /*
+    ┌──────────────────┐         ┌───────────────────┐
+    │      plugin      │         │  plugin_version   │
+    ├──────────────────┤         ├───────────────────┤
+    │public_id (pk)    │        ╱│public_id (pk)     │
+    │scope_id (fk)     │┼┼────┼──│plugin_id (fk)     │
+    └──────────────────┘        ╲│semantic_version   │
+                                 └───────────────────┘
+                                           ┼
+                                           ┼
+                                           │
+                                           ┼
+                                          ╱│╲
+                               ┌──────────────────────┐
+                               │  plugin_executable   │
+                               ├──────────────────────┤
+                               │version_id (pk, fk)   │
+                               │operating_system (pk) │
+                               │architecture (pk)     │
+                               │executable            │
+                               └──────────────────────┘
+   */
+  create table plugin_version (
+    public_id wt_public_id primary key,
+    plugin_id wt_public_id not null
+      constraint plugin_fkey
+        references plugin (public_id)
+        on delete cascade
+        on update cascade,
+    semantic_version text not null
+      constraint plugin_version_requires_semantic_version
+      check(length(semantic_version) > 4), -- minimum length is length("0.0.0")
+    create_time wt_timestamp,
+
+    unique(plugin_id, public_id),
+    unique(plugin_id, semantic_version)
+  );
+
+  create trigger default_create_time_column before insert on plugin_version
+    for each row execute procedure default_create_time();
+
+  create trigger immutable_columns before update on plugin_version
+    for each row execute procedure immutable_columns('public_id', 'plugin_id', 'create_time', 'semantic_version');
+
+  create table plugin_executable (
```

Comment:
Is the executable supposed to be encrypted?

Reply:
removed the executable table.